native/src/main/java/io/ballerina/stdlib/java.jms/listener/Service.java

```diff
@@ -83,17 +95,22 @@ public static void validateService(BObject consumerService) throws BError {
         }
 
         RemoteMethodType[] remoteMethods = service.getRemoteMethods();
-        if (remoteMethods.length != 1) {
-            throw CommonUtils.createError(JMS_ERROR, "JMS service must have exactly one remote method.");
+        if (remoteMethods.length < 1 || remoteMethods.length > 2) {
+            throw CommonUtils.createError(
+                    JMS_ERROR, "JMS service must have exactly one or two remote methods.");
         }
 
-        RemoteMethodType existingRemoteMethod = remoteMethods[0];
-        if (!existingRemoteMethod.getName().equals("onMessage")) {
-            throw CommonUtils.createError(JMS_ERROR,
-                    "JMS service does not contain the required 'onMessage' method.");
+        for (RemoteMethodType remoteMethod: remoteMethods) {
+            String remoteMethodName = remoteMethod.getName();
+            if (ON_MSG_METHOD.equals(remoteMethodName)) {
+                validateOnMessageMethod(remoteMethod);
+            } else if (ON_ERR_METHOD.equals(remoteMethodName)) {
+                validateOnErrorMethod(remoteMethod);
+            } else {
+                throw CommonUtils.createError(
+                        JMS_ERROR, String.format("Invalid remote method name: %s.", remoteMethodName));
+            }
```

Comment:
Since we are anyway iterating through the remote methods here, can't we just set the class variables here too? Just a thought, a minor detail.